rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/ChargingRSpace.scala

```diff
@@ -39,14 +41,24 @@ object ChargingRSpace {
           patterns: Seq[BindPattern],
           continuation: TaggedContinuation,
           persist: Boolean,
-          sequenceNumber: Int
+          sequenceNumber: Int,
+          peek: Boolean
```

Comment:
The flag "peek" which selects different behaviours is a bad smell.

Another issue is that the name of the method "consume" which is no longer valid if we are talking about peeks. 

I think this can be fixed by having two methods: `consume` and `peek` in RhoPureSpace

Reply:
This might make sense temporarily, but if we extend Rholang syntax to allow mixing standard/persistent/peek reads in binds under the same receive this change will have to be reverted

Reply:
I understand. So we need one method to cover a mixture of peek and consumes. Still, the flag is a bad smell as we should not pass a flag to switch between the two behaviours internally. 

What if we pass the list of "peeked channels" and the list of "consumed channels"? This way we won't need the flag